src/osd/PG.cc

```diff
@@ -4645,7 +4693,13 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
           // will be requeued by sub_op_scrub_map
           dout(10) << "wait for replicas to build scrub map" << dendl;
           done = true;
-        } else {
+        } else if (scrubber.preempted) {
+	  scrubber.preempt_left--;
+	  scrubber.preempt_divisor *= 2;
+	  dout(10) << __func__ << " preempted, " << scrubber.preempt_left
+		   << " left, restarting chunk" << dendl;
+	  scrubber.state = PG::Scrubber::NEW_CHUNK;
+	} else {
           scrubber.state = PG::Scrubber::COMPARE_MAPS;
```

Comment:
what about reset scrubber.preempt_left and scrubber.preempt_divisor to their default value(5 and 1) after COMPARE_MAPS and before begin a NEW_CHUNK, becasue pg may have log objects. every scrub chunk has its new preempt_left may better.